### PR TITLE
Update eclipselink from 2.7.9 to 2.7.10

### PR DESCRIPTION
**Pull request by pepness**
Library Notes:

- This is mainly a maintenance release and contains bug fixes
- Use jakarta.persistence instead of javax.persistence (still using javax.* in packages)
- Deprecated Functionality: None

NetBeans Testing:
- Full build done
- Verify successful execution of libraries and licenses Ant test
- Verify successful execution of unit tests for modules `websvc.restlib`, `j2ee.eclipselink`, `j2ee.eclipselinkmodelgen`, `j2ee.jpa.refactoring`, `j2ee.metadata.model.support`, `j2ee.persistence` and `j2ee.persistenceapi`
- Started NetBeans and ensure the log didn't have any ERROR or new WARNINGS
- Checked the files are in the Ant Library Manager
- Successfully create project as per [NetBeans Documentation](https://netbeans.apache.org/kb/docs/websvc/rest.html), with Payara and Glassfish

[EclipseLink Web Page](https://www.eclipse.org/eclipselink/)
[EclipseLink Release Notes](https://github.com/eclipse-ee4j/eclipselink/releases/tag/2.7.10)

### CODE REVIEW

File: java/j2ee.eclipselink/nbproject/project.xml [hunk 1]
```diff
@@ -26,29 +26,126 @@
             <code-name-base>org.netbeans.modules.j2ee.eclipselink</code-name-base>
             <module-dependencies/>
             <public-packages>
-                <subpackages>org.eclipse</subpackages>
+                <subpackages>org.eclipse.persistence</subpackages>
```
Comment (matthiasblaesing) [inline java/j2ee.eclipselink/nbproject/project.xml:29]: What is the reason for this change? Are there internal packages, that shall be hidden? If only a single package should be exposed, I would expect `package`, `subpackages` matches all packages where the given name is a prefix.

**Review by matthiasblaesing — APPROVED** [review]
In general this looks good. There is one change, that I don't understand, please see inline.

**Comment by mbien** [comment]
sigtests are failing on this PR even after restart.

File: java/j2ee.eclipselink/nbproject/project.xml (hunk 1 above)
Comment (pepness) [reply java/j2ee.eclipselink/nbproject/project.xml:29]: I was under the impression that I should only include `subpackages` that have java classes inside and exclude internal packages like `org.eclipse.persistence.internal`. What would be the proper way of configure `<public-packages>` and `<subpackages>`, include all packages excluding internal packages?

Comment (matthiasblaesing) [reply java/j2ee.eclipselink/nbproject/project.xml:29]: If you want to expose only the package and not the package where it is a prefix of, `package` is the right element. Your assessment is right, internal should not be exposed.

The sigtest fails from my interpretation because classes are missing:

https://app.travis-ci.com/github/apache/netbeans/jobs/560459188#L12806-L12808

This smells like these:
https://search.maven.org/artifact/jakarta.resource/jakarta.resource-api/1.7.4/jar
https://search.maven.org/artifact/jakarta.transaction/jakarta.transaction-api/1.3.3/jar

Comment (pepness) [reply java/j2ee.eclipselink/nbproject/project.xml]: I restore the entry `<subpackages>org.eclipse</subpackages>`, deleted the added entries, regenerate sigtest and everything worked fine. I suppose that if another module need a class from this packages it would fail the build process.

**Comment by mbien** [comment]
tests are green, PR is approved / no outstanding change requests as far as I see -> merging.
